_posts/2020-08-19-#19509.md

```diff
@@ -0,0 +1,53 @@
+---
+layout: pr
+date: 2020-08-19
+title: "Per-peer Message Logging"
+pr: 19509
+authors: [troygiorshev]
+components: ["p2p", "utils/log/libs"]
+host: troygiorshev
+status: upcoming
+commit: 4fc2e3906
+---
+
+## Notes
+
+* [PR 19509](https://github.com/bitcoin/bitcoin/pull/19509) is a feature PR, adding per-peer message logging to Bitcoin Core.
+
+* Nodes on the Bitcoin network communicate using a variety of [messages](https://btcinformation.org/en/developer-reference#p2p-network).  Now these messages can be logged for easy review, in order and on a per-peer basis.
+
+* The purpose of this feature is twofold: it should assist people in debugging their node, and it should help people who are unfamiliar with the Bitcoin network to get a feel of how everything works.
+
+* This feature is locked behind a debug-only `-logmessages` option.  When enabled, it writes every message the node recieves to a file, in a folder structure described in the PR.  It uses our `CAutoFile` [here](https://github.com/bitcoin/bitcoin/blob/ac953a3773aa5e940976a79276894f5e892d1c55/src/net.cpp#L2892).
+
+* The log parser lives in a new folder, at `contrib/message-logging/message-logging-parser.py`.  As discussed [here](https://github.com/bitcoin/bitcoin/pull/19509#issuecomment-664374225) it takes a list of files to parse, and optionally an output argument.
+
+* `LogMessage` takes the newly introduced `Span`, as discussed in a [previous PR review club](https://bitcoincore.reviews/18468).
+
+* [This commit](https://github.com/bitcoin/bitcoin/pull/19509/commits/ac953a3773aa5e940976a79276894f5e892d1c55) accounts for the increase in file descriptor usage.  This is due to be improved in [Issue 18911](https://github.com/bitcoin/bitcoin/issues/18911).
```

Comment:
would suggest:

- s/this commit/commit ac953a3/ both here and line 37

- and maybe adding the titles of the commit and the issue so people can see their names without having to click through on first read

Reply:
I've changed all commit links to explicitly name the commits they're linking to.